tests/ci/ci.py

```diff
@@ -48,6 +52,561 @@
+
+    def __init__(
+        self,
+        s3: S3Helper,
+        job_digests: Dict[str, str],
+    ):
+        self.s3 = s3
+        # self.build_digest = job_digests[JobNames.PACKAGE_RELEASE]
+        # self.docs_digest = job_digests[JobNames.DOCS_CHECK]
+        self.job_digests = job_digests
+        # self.build_path = f"{self._S3_CACHE_PREFIX}/BUILD-{self.build_digest}/"
+        # self.docs_path = f"{self._S3_CACHE_PREFIX}/DOCS-{self.docs_digest}/"
+        self.cache_s3_paths = {
+            job_type: f"{self._S3_CACHE_PREFIX}/{job_type.value}-{self.job_digests[self._get_reference_job_name(job_type)]}/"
+            for job_type in self.JobType
+        }
+        self.s3_record_prefixes = {
+            record_type: record_type.value for record_type in self.RecordType
+        }
+        self.records: Dict["CiCache.RecordType", Dict[str, "CiCache.Record"]] = {
+            record_type: {} for record_type in self.RecordType
+        }
+
+        self.cache_updated = False
+        self.cache_data_fetched = True
+        if not self._LOCAL_CACHE_PATH.exists():
+            self._LOCAL_CACHE_PATH.mkdir(parents=True, exist_ok=True)
+
+    def _get_reference_job_name(self, job_type: JobType) -> str:
+        res = Build.PACKAGE_RELEASE
+        if job_type == self.JobType.DOCS:
+            res = JobNames.DOCS_CHECK
+        elif job_type == self.JobType.SRCS:
+            res = Build.PACKAGE_RELEASE
+        else:
+            assert False
+        return res
+
+    def _get_record_file_name(
+        self,
+        record_type: RecordType,
+        job_name: str,
+        batch: int,
+        num_batches: int,
+        release_branch: bool,
+    ) -> str:
+        prefix = self.s3_record_prefixes[record_type]
+        prefix_extended = (
+            self._DIV2.join([prefix, self._ATTRIBUTE_RELEASE])
+            if release_branch
+            else prefix
+        )
+        job_name = self._DIV2.join(
+            [job_name, self.job_digests[job_name], str(batch), str(num_batches)]
+        )
+        file_name = self._DIV1.join([prefix_extended, job_name])
+        file_name += self._RECORD_FILE_EXTENSION
+        return file_name
+
+    def _get_record_s3_path(self, job_name: str) -> str:
+        return self.cache_s3_paths[self.JobType.get_type_by_name(job_name)]
+
+    def update(self):
+        """
+        Pulls cache records from s3. Only records name w/o content.
+        """
+        for record_type in self.RecordType:
+            prefix = self.s3_record_prefixes[record_type]
+            cache_list = self.records[record_type]
+            for job_type in self.JobType:
+                path = self.cache_s3_paths[job_type]
+                records = self.s3.list_prefix(f"{path}{prefix}", S3_BUILDS_BUCKET)
+                records = [record.split("/")[-1] for record in records]
+                print(f"::group::Cache records for {job_type.value} type jobs")
+                print(records)
+                print("::endgroup::")
```

Comment:
It already looks like a dedicated function to me.

`actions_log_group(group_name: str, lines: str)`